tools/wutool/wutool.cpp

```diff
@@ -1570,6 +1571,81 @@ class WuTool : public CppUnit::TestFixture
         ASSERT(numIterated == before);
     }
 
+    void testList3()
+    {
+        // Now by owner and 2 states via generic mechanism
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+        unsigned start = msTick();
+        Owned<IConstWorkUnitIterator> wus;
+        __int64 cachehint = 0;
+        unsigned startRow = 0;
+
+        unsigned numCompleted = 0;
+        unsigned numFailed = 0;
+        WUSortField sortByOwner[] = { WUSFuser, WUSFstate, WUSFterm };
+
+        unsigned before = factory->numWorkUnits();
+        if (before > 0)
+        {
+            start = msTick();
+            wus.setown(factory->getWorkUnitsSorted((WUSortField) (WUSFwuid | WUSFreverse), sortByOwner, "WuTestUser00\0completed|failed", 0, 10000, NULL, NULL));
+            ForEach(*wus)
+            {
+                IConstWorkUnitInfo &wu = wus->query();
+                ASSERT(streq(wu.queryUser(), "WuTestUser00"));
+                ASSERT((wu.getState()==WUStateCompleted) || (wu.getState()==WUStateFailed));
+                if ((wu.getState()==WUStateCompleted))
+                    numCompleted++;
+                else
+                    numFailed++;
+            }
+            DBGLOG("%d completed workunits listed and %d failed workunits listed in %d ms", numCompleted, numFailed, msTick()-start);
+        }
+
+        if ((numCompleted > 0) || (numFailed > 0))
```

Comment:
Not sure what you are trying to achieve here - the tests in wutool should assume that the repository being tested is in a known state (at least as far as the fake workunits created by this test are concerned).